src/Symfony/Component/Translation/Translatable/DecimalTranslatable.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Translatable;
+
+use Symfony\Contracts\Translation\TranslatableInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Wrapper around PHP NumberFormatter for decimal values.
+ *
+ * Implementation of the ICU recommendation to first format advanced parameters before translation.
+ *
+ * @see https://unicode-org.github.io/icu/userguide/format_parse/messages/#format-the-parameters-separately-recommended
+ *
+ * @author Sylvain Fabre <syl.fabre@gmail.com>
+ */
+class DecimalTranslatable implements TranslatableInterface
+{
+    private $style;
+    private $value;
+
+    private $formatters = [];
+
+    public function __construct($value, int $style = \NumberFormatter::DECIMAL)
```

Comment:
missing type for `$value` (when it cannot be done using a native type, phpdoc must be used, but our public API must define its types)